### PR TITLE
fix(curriculum): make expected result format in instructions and tests match

### LINKED ISSUE #40763: Rosetta Code - Zeckendorf number representation - test doesn't match task in challenge description
**Describe your problem and how to reproduce it:**
Challenge description asks for an **array of the first n** Zeckendorf number representations in order. On the other hand second test expects function to return just a single representation of **number n**:
```
assert.deepEqual(answer, solution20);
```
And those variables from `--after-user-code--`:
```
const range = (m, n) => (
  Array.from({
    length: Math.floor(n - m) + 1
  }, (_, i) => m + i)
);

const solution20 = [
  '1', '10', '100', '101', '1000', '1001', '1010', '10000', '10001',
  '10010', '10100', '10101', '100000', '100001', '100010', '100100', '100101',
  '101000', '101001', '101010'
];

const answer = range(1, 20).map(zeckendorf);
```
Description of the relevant test is very general and it wouldn't be affected when clearing above inconsistency, However it could say something more than just that answer needs to be correct.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/zeckendorf-number-representation
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/zeckendorf-number-representation.md

### PR DESCRIPTION
**Pull request by gikf**
- Unifies instructions and tests to require calculating representation for `n`.
- Changes sample solution to return integer instead of string.
- Previously challenge calculated, in the background, representation of numbers up to `20` (inclusive), I've changed that to individual tests. Doesn't that make too many tests?
- Partially related to #40896.
- Tested on local fork.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40763

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/zeckendorf-number-representation.md [hunk 1]
```diff
@@ -14,42 +14,145 @@ The decimal number eleven can be written as `0*13 + 1*8 + 0*5 + 1*3 + 0*2 + 0*1`
 
 # --instructions--
 
-Write a function that generates and returns an array of the first `n` Zeckendorf numbers in order.
+Write a function that generates and returns Zeckendorf number representation of `n`.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/zeckendorf-number-representation.md:17]:
```suggestion
Write a function that generates and returns the Zeckendorf number representation of `n`.
```

**Review by moT01 — COMMENTED** [review]
Minor suggestion. Other than that, LGTM 🎉

**Comment by moT01** [comment]
Hey @gikf, we have a [chat server](https://chat.freecodecamp.org/) where we discuss things if you want to join.

**Comment by gikf** [comment]
@moT01 :+1:  I'm there, but with quite different username - sanity

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Review by Sembauke — APPROVED** [review]
LGTM  🎉

**Pull request merged**